src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -160,6 +159,28 @@
         ),
         Ref("AliasExpressionSegment", optional=True),
     ),
+    ColumnReferenceSegment=Delimited(
+        Sequence(
+            ansi.ColumnReferenceSegment,
```

Comment:
Have we used this technique elsewhere?

One question (and possible concern): Are we overriding a segment definition with a grammar? Could that cause issues?

Reply:
Postgres used it. Didn’t like it originally but see advantages to it now for things like this!

Reply:
Any more thoughts on this @barrywhart ? It's been in Postgres for 6 months for the exact same segment since #1485 was merged and there were no issues AFAIK.

Reply:
Nope, if it works, I'm good. I don't know the parser very well.